csharp/ql/lib/semmle/code/cil/Stubs.qll

```diff
@@ -29,14 +29,17 @@ private module Cached {
 
   cached
   predicate bestImplementation(MethodImplementation mi) {
-    not assemblyIsStubImpl(mi.getLocation()) and
-    not exists(MethodImplementation better | mi.getMethod() = better.getMethod() |
-      mi.getNumberOfInstructions() < better.getNumberOfInstructions()
-      or
-      mi.getNumberOfInstructions() = better.getNumberOfInstructions() and
-      mi.getLocation().getFile().toString() > better.getLocation().getFile().toString()
-    ) and
-    exists(mi.getAnInstruction())
+    exists(Assembly asm |
+      asm = mi.getLocation() and
+      (assemblyIsStubImpl(asm) implies asm.getFile().extractedQlTest()) and
```

Comment:
Why are assembly stub implementations taken into considerations now (looks like it is relevant for test)?

Reply:
Yeah, this is only relevant for tests. One of our test assemblies would otherwise incorrectly be considered a stub, and therefore not be included in `TCilExprNode`.